### PR TITLE
[SYCL][Test E2E] Use %{build}/%{run} in miscellaneous tests, part 2

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: sycl/test-e2e/KernelAndProgram/kernel-bundle-merge-options.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// RUN: %{build} -o %t.out %debug_option
-// RUN: env SYCL_PI_TRACE=-1 %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
 // REQUIRES: gpu
+// RUN: %{build} -o %t.out %debug_option
+// RUN: env SYCL_PI_TRACE=-1 %{run} %t.out | FileChe%s
```
Comment (aelovikov-intel) [inline sycl/test-e2e/KernelAndProgram/kernel-bundle-merge-options.cpp:3]:
```suggestion
// RUN: env SYCL_PI_TRACE=-1 %{run} %t.out | FileCheck %s
```

**Review by cperkinsintel — APPROVED** [review]
LGTM

File: sycl/test-e2e/DeviceLib/assert-aot.cpp [hunk 2]
```diff
@@ -1,5 +1,4 @@
 // REQUIRES: opencl-aot, cpu, linux
 
 // RUN: %clangxx -DSYCL_FALLBACK_ASSERT=1 -fsycl -fsycl-targets=spir64_x86_64 %S/assert.cpp -o %t.aot.out
-// RUN: env EXPECTED_SIGNAL=SIGABRT SHOULD_CRASH=1 %CPU_RUN_PLACEHOLDER %t.aot.out 2>%t.aot.msg
-// RUN: FileCheck %S/assert.cpp --input-file %t.aot.msg --check-prefixes=CHECK-MESSAGE
+// RUN: env EXPECTED_SIGNAL=SIGABRT SHOULD_CRASH=1 %{run} %t.aot.out 2>&1 | FileCheck %S/assert.cpp --check-prefixes=CHECK-MESSAGE
```
[the remark this answers is not in the document]
Comment (aelovikov-intel) [reply sycl/test-e2e/DeviceLib/assert-aot.cpp:4]: No, but close enough. The difference is that the `2>%t.aot.msg` only captures `stderr` while `2>&1` captures both `stdout`/`stderr`.

**Comment by dm-vodopyanov** [comment]
Issues for the failure in post-commit: https://github.com/intel/llvm/issues/9387

**Pull request merged**